### PR TITLE
Fixes #11031 - ignore identifiers of interfaces being destroyed

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

File: test/unit/host_test.rb [hunk 1]
```diff
@@ -2335,6 +2335,30 @@ def to_managed!
...
+    let(:host) { FactoryGirl.build(:host, :managed) }
+    let(:additional_interface) { host.interfaces.build }
+
+    context 'additional interface has different identifier' do
+      test 'host is valid' do
+        assert host.valid?
+      end
+    end
+
+    context 'additional interface has different identifier' do
```
Comment (domcleal) [inline test/unit/host_test.rb]: same identifier

File: app/models/host/base.rb [hunk 2]
```diff
@@ -426,7 +426,8 @@ def check_provision_interface
     def uniq_interfaces_identifiers
       success = true
       identifiers = []
-      self.interfaces.each do |interface|
+      relevant_interfaces = self.interfaces.select { |i| !i._destroy }
```
Comment (domcleal) [inline app/models/host/base.rb]: Nitpick, would `marked_for_destruction?` from autosave_association be better?  I realise they amount to the same thing, it's just _destroy seems like it's meant only for use with nested attrs.

**Comment by ares** [comment]
Fixed both, I didn't realize that accepts_nested_attributes_for sets autosave to true, thanks.

File: test/unit/host_test.rb [hunk 3]
```diff
@@ -2335,6 +2335,30 @@ def to_managed!
     end
   end
 
+  describe 'interface identifiers validation' do
```
Comment (dLobatog) [inline test/unit/host_test.rb:2338]: These tests read beautifully :sparkles:

**Comment by dLobatog** [comment]
Merged as eaf61bb1751fd6287830f0ade8967236e4fcc240, thanks @ares!

**Pull request closed without merging**